pycode/memilio-epidata/memilio/epidata/getDIVIData.py

```diff
@@ -168,23 +218,94 @@ def get_divi_data(read_data=dd.defaultDict['read_data'],
+        gd.write_dataframe(df_ger, directory, filename, file_format)
 
+    return df_counties, df_states, df_ger
 
-def divi_data_sanity_checks(df=pd.DataFrame()):
+
+def get_divi_data(read_data: bool = dd.defaultDict['read_data'],
+                  file_format: str = dd.defaultDict['file_format'],
+                  out_folder: str = dd.defaultDict['out_folder'],
+                  start_date: date = date(2020, 4, 24),
+                  end_date: date = dd.defaultDict['end_date'],
+                  impute_dates: bool = dd.defaultDict['impute_dates'],
+                  moving_average: int = dd.defaultDict['moving_average'],
+                  to_dataset: bool = dd.defaultDict['to_dataset'],
+                  **kwargs
+                  ):
+    """! Downloads or reads the DIVI ICU data and writes them in different files.
+
+    Available data starts from 2020-04-24.
+    If the given start_date is earlier, it is changed to this date and a warning is printed.
+    If it does not already exist, the folder Germany is generated in the given out_folder.
+    If read_data == True and the file "FullData_DIVI.json" exists, the data is read form this file
+    and stored in a pandas dataframe. If read_data = True and the file does not exist the program is stopped.
+
+    The downloaded dataframe is written to the file "FullData_DIVI".
+    After that, the columns are renamed to English and the state and county names are added.
+    Afterwards, three kinds of structuring of the data are done.
+    We obtain the chronological sequence of ICU and ICU_ventilated
+    stored in the files "county_divi".json", "state_divi.json" and "germany_divi.json"
+    for counties, states and whole Germany, respectively.
+
+    @param read_data True or False. Defines if data is read from file or downloaded. Default defined in defaultDict.
+    @param file_format File format which is used for writing the data. Default defined in defaultDict.
+    @param out_folder Folder where data is written to. Default defined in defaultDict.
+    @param start_date Date of first date in dataframe. Default defined in defaultDict.
+    @param end_date Date of last date in dataframe. Default defined in defaultDict.
+    @param impute_dates True or False. Defines if values for dates without new information are imputed. Default defined in defaultDict.
+    @param moving_average Integers >=0. Applies an 'moving_average'-days moving average on all time series
+        to smooth out effects of irregular reporting. Default defined in defaultDict.
+    @param to_dataset bool True or False. Whether to return the dataframe as an object instead of json file.
+        If True - returns objects with dataframes
+        If False - write dataframes into files
+        Default defined in defaultDict.
+    """
+    downloaded_data_df = fetch_divi_data(
+        read_data=read_data,
+        out_folder=out_folder,
+        file_format=file_format,
+        start_date=start_date,
+        **kwargs
```

Comment:
Same as in getCaseDate above. Call conf once and then there is no need to use`**kwargs`.